{
  if (response.isCommitted()) {
    logger.log(Level.WARNING,"Caught exception, but response already commited. Not sending error message based on exception",e);
  }
 else {
    String message;
    if (e instanceof AccessDeniedException || e instanceof BadCredentialsException) {
      message="{\"status\":\"error\", \"error\":\"" + CloudifyErrorMessages.NO_PERMISSION_ACCESS_DENIED.getName() + "\"}";
      logger.log(Level.INFO,e.getMessage(),e);
    }
 else {
      logger.log(Level.WARNING,"An unexpected error was thrown: " + e.getMessage(),e);
      final Map<String,Object> restErrorMap=RestUtils.verboseErrorStatus(CloudifyErrorMessages.GENERAL_SERVER_ERROR.getName(),ExceptionUtils.getStackTrace(e),e.getMessage());
      message=new ObjectMapper().writeValueAsString(restErrorMap);
    }
    final ServletOutputStream outputStream=response.getOutputStream();
    final byte[] messageBytes=message.getBytes();
    outputStream.write(messageBytes);
  }
}
